activerecord/lib/active_record/encryption/configurable.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module Encryption
+    # Configuration API for +ActiveRecord::Encryption+
+    module Configurable
+      extend ActiveSupport::Concern
+
+      included do
+        mattr_reader :config, default: Config.new
+        mattr_accessor :encrypted_attribute_declaration_listeners
+      end
+
+      class_methods do
+        # Expose getters for context properties
+        Context::PROPERTIES.each do |name|
+          delegate name, to: :context
+        end
+
+        def configure(master_key:, deterministic_key:, key_derivation_salt:, **properties) #:nodoc:
+          config.master_key = master_key
+          config.deterministic_key = deterministic_key
+          config.key_derivation_salt = key_derivation_salt
+
+          context.key_provider = ActiveRecord::Encryption::DerivedSecretKeyProvider.new(master_key)
+
+          properties.each do |name, value|
+            [:context, :config].each do |configurable_object_name|
+              configurable_object = ActiveRecord::Encryption.send(configurable_object_name)
+              configurable_object.send "#{name}=", value if configurable_object.respond_to?(name)
+            end
+          end
+        end
+
+        # Register callback to be invoked when an encrypted attribute is declared.
+        #
+        # === Example:
+        #
+        #   ActiveRecord::Encryption.on_encrypted_attribute_declared do |klass, attribute_name|
+        #     ...
+        #   end
+        def on_encrypted_attribute_declared(&block)
```

Comment:
This method can be called when loading model classes that include a `.encrypts` declaration. In production, with eager loading, it should happen only once, but in development, with lazy-loading, I think it might happen that two threads access concurrently. E.g: with async job adapter, with 2 concurrent web requests. I think the thread safety there is fine.